### ISSUE TITLE
Transcation

### ISSUE THREAD

**Issue by jayrajk** [szokodiakos/typegoose#258]
Add on functionality of transaction which mongoose has provide
https://mongoosejs.com/docs/transactions.html

**Comment by hasezoey** [comment]
i know this is a bit old, but:
why would this be needed to be included in typegoose, because as far as i could see, this has nothing todo with the model creation / schema creation?

**Comment by hasezoey** [comment]
I'm currently investigating this for release 6.0.0, but it seems like this cannot/should not be implemented in typegoose, here are the reasons:
- `const session = mongoose.startSession()` and the following `session.startTransaction()` have nothing to do with typegoose
- it cannot be tested without changing the tests to use (package)`run-rs`
- it works (with types and co) without Typegoose implementing it

full example (needs Replica Set connection):
```ts
const session = await mongoose.startSession();
session.startTransaction();
await somemodel.create({}, { session });
await somemodel.find({}, { session });
await session.commitTransaction():
```

TL;DR: "wontfix" & removing from 6.0.0 spec (this can be closed)
@Ben305

**Comment by hasezoey** [comment]
@Ben305 Bump to close

**Comment by vanishdark** [comment]
Hey, maybe coming a lit late, but just to people arrive here looking for typegoose transactions, i have an example may you can use.

```
export class ClassName {
           constructor (@InjectModel('modelname') private readonly modelname: ReturnModelType<typeof ModelClass>)
}

public findOneAndReplaceByOne(){
 let session: ClientSession = null;
        this.modelname.db.startSession()
        .then(_session => {
            session = _session;
            session.startTransaction();
            return this.modelname.findOne({});
        })
        .then((model: any) => {
            model.ENUMID = model.ENUMID + 1;
            return refData.save();
        })
        .then((result) => {
                session.commitTransaction();
            })
        .catch(e => {
               session.abortTransaction();
               console.error(e);
});
}
```

ofc you can turn this into async await., just follow the same logic

**Comment by sebastiangug** [comment]
@vanishdark 

why are we using the object reference in the constructor when reading/modifying data? 

If I have other unrelated queries that use the same object and then I abort my transaction, won't that abort all other operations as well?
